src/components/molecules/WalkThrough/types.ts

```diff
@@ -1,10 +1,12 @@
+export type WalkThroughCollection = 'novice' | 'release';
 export type WalkThroughStep = 'template' | 'resource' | 'syntax' | 'cluster' | 'kustomizeHelm' | 'validation';
```

Comment:
A problem is that we can mix and match collections and steps in incorrect combinations. For example, you can set `<Walkthrough collection="release" step="template" />` which is not supported. Fixing it is a three step process.

1. Make step conditional based on the collection:
```
export type WalkThroughStep<C extends WalkThroughCollection = WalkThroughCollection> = C extends 'novice'
  ? 'template' | 'resource' | 'syntax' | 'cluster' | 'kustomizeHelm'
  : 'validation';
```
The default is there to avoid it complains everywhere but it gives you the option to get a narrowed result when you do add the collection.


2. Make a connection between the passed collection and steps:
```
// src/components/molecules/WalkThrough/WalkThrough.tsx:16
type WalkThroughProps<C extends WalkThroughCollection> = {
  step: WalkThroughStep<C>;
  children: React.ReactNode;
  placement?: PopoverProps['placement'];
  collection: C;
};
```

3. Allow the prop to be inferred automatically:
```
// src/components/molecules/WalkThrough/WalkThrough.tsx:71
const WalkThrough = <C extends WalkThroughCollection>(props: WalkThroughProps<C>) => {
```

4. Validate that it works. API consumption does not require declaring the generic as it can infer this.
```
//src/components/organisms/NavigatorPane/ClusterCompareButton.tsx:31~32
// error - Type '"validation"' is not assignable to type '"template" | "resource" | "syntax" | "cluster" | "kustomizeHelm"'.
 <WalkThrough placement="leftTop" step="validation" collection="novice">

```

Reply:
@WitoDelnat, good point! I've updated types